android/java/org/chromium/chrome/browser/crypto_wallet/fragments/dapps/SiweMessageFragment.java

```diff
@@ -5,23 +5,293 @@
+                    } else if (accountAddress != null) {
+                        Utils.setTextGeneratedBlockies(
+                                mExecutor, mHandler, mIvFav, accountAddress, true);
+                    }
+                });
+    }
+
+    private void updateNetwork(String chainId) {
+        if (JavaUtils.anyNull(mWalletModel, chainId)) return;
+        NetworkInfo network = mWalletModel.getNetworkModel().getNetwork(chainId);
+        if (network == null) return;
+        String tokensPath = BlockchainRegistryFactory.getInstance().getTokensIconsLocation();
+        String logo = Utils.getNetworkIconName(network.chainId, network.coin);
+        if (TextUtils.isEmpty(logo)) return;
+        Utils.setBitmapResource(mExecutor, mHandler, requireContext(),
+                "file://" + tokensPath + "/" + logo, Integer.MIN_VALUE, mIvFavNetwork, null, true);
+        AndroidUtils.show(mIvFavNetworkContainer);
+        mNetworkName.setText(network.chainName);
+    }
+
+    private void showSiweDetails() {
+        assert mCurrentSignMessageRequest.signData.which() == SignDataUnion.Tag.EthSiweData;
+
+        List<TwoLineItemRecyclerViewAdapter.TwoLineItem> items = new ArrayList<>();
+        addDetail(items, R.string.brave_wallet_origin, getOriginJson(mSiweMessageData.origin));
+        addDetail(items, R.string.brave_wallet_address, mSiweMessageData.address);
+        addDetail(items, R.string.brave_wallet_statement, mSiweMessageData.statement);
+        addDetail(items, R.string.brave_wallet_uri, mSiweMessageData.uri.url);
+        addDetail(items, R.string.brave_wallet_version, Integer.toString(mSiweMessageData.version));
+        addDetail(items, R.string.brave_wallet_chain_id, Long.toString(mSiweMessageData.chainId));
+        addDetail(items, R.string.brave_wallet_nonce, mSiweMessageData.nonce);
+        addDetail(items, R.string.brave_wallet_issued_at, mSiweMessageData.issuedAt);
+        addDetail(items, R.string.brave_wallet_expiration_time, mSiweMessageData.expirationTime);
+        TwoLineItemBottomSheetFragment fragment = TwoLineItemBottomSheetFragment.newInstance(items);
+        fragment.mTitle = getString(R.string.brave_wallet_see_details);
+        fragment.show(getParentFragmentManager(), TAG);
+    }
+
+    private void addDetail(List<TwoLineItemRecyclerViewAdapter.TwoLineItem> allDetails,
+            @StringRes int id, String value) {
+        if (TextUtils.isEmpty(value)) return;
+        allDetails.add(new TwoLineItemRecyclerViewAdapter.TwoLineItemText(getString(id), value));
+        allDetails.add(new TwoLineItemRecyclerViewAdapter.TwoLineItemDivider());
+    }
+
+    private String getOriginJson(org.chromium.url.internal.mojom.Origin origin) {
+        if (origin == null) return null;
+
+        try {
+            JSONObject jsonObject = new JSONObject();
+
+            jsonObject.put("scheme", origin.scheme);
+            jsonObject.put("host", origin.host);
+            jsonObject.put("port", origin.port);
```

Comment:
Info: LMK if we should also localise these URL titles (scheme, host, port).

Reply:
I think we are fine with how it is now, but I will let string reviewers to comment on it.